### PR TITLE
refactor(schemas): Updated schemas with enum values

### LINKED ISSUE #18: Use enum values in JSON Schema
Some of the fields that we are restricting as strings can be expressed by enums for better suggestions and better validation rules:
- globals.assets.type (link, script, style, etc.)
- globals.manifest - multiple fields (check specification)
- dependency.type (local, package, library)
- propDefinitions.type
- stateDefinitions.type

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
fixes #18

### CODE REVIEW

**Comment by JayaKrishnaNamburu** [comment]
Hi, @alexnm I have updated the schema attributes using enum, please let me know if any further changes/corrections needed.

**Comment by JayaKrishnaNamburu** [comment]
HI @alexnm i am a little new to typescript so for assigning ``enum`` for ``global.asset.type`` there are two ways, 

First either define an enum and cross check with that enum in all test cases like
~~~
export enum AssetType {
  "style",
}
if (asset.type === AssetType.style)
~~~
or declaring the ``asset.type`` in the following way
~~~
type: "style" | "script"
~~~

Please let me know which helps our use case here 😅

File: src/uidl-definitions/schemas/component.json [hunk 1]
```diff
@@ -200,7 +204,7 @@
           "additionalProperties": false,
           "required": ["type", "defaultValue"],
           "properties": {
-            "type": {"type": "string"},
+            "type": {"type":"string", "enum": ["string", "boolean", "number", "object"]},
```
Comment (alexnm) [inline src/uidl-definitions/schemas/component.json]: same thing is needed for the type inside `propDefinition`
and you should also include `array` and `function`

**Comment by alexnm** [comment]
@JayaKrishnaNamburu actually I gave it a bit more thought and it makes more sense to use the enums like you proposed for the asset type. the advantage with that is that we won't be writing strings in the code anymore if we have the enums. do you agree?

File: src/uidl-definitions/types/index.ts [hunk 2]
```diff
@@ -120,6 +128,7 @@ export interface WebManifest {
   orientation?: string
   scope?: string
   theme_color?: string
+  screenshots?: Array<{ src: string; sizes: string; type: string }>
```
Comment (alexnm) [inline src/uidl-definitions/types/index.ts]: What does this represent?

File: src/uidl-definitions/types/index.ts [hunk 3]
```diff
@@ -15,8 +15,16 @@ export interface ProjectUIDL {
   components?: Record<string, ComponentUIDL>
 }
 
+enum AssetType {
```
Comment (alexnm) [inline src/uidl-definitions/types/index.ts]: these changes should also be reflected in the schema, not only in the TS interfaces

**Comment by JayaKrishnaNamburu** [comment]
> @JayaKrishnaNamburu actually I gave it a bit more thought and it makes more sense to use the enums like you proposed for the asset type. the advantage with that is that we won't be writing strings in the code anymore if we have the enums. do you agree?

Yes, I took agree with the point of not using strings anywhere in the codebase but was not sure with the scope of that implementation other than that use case. Personally I like to keep it this way but we need to import and use that ``enum`` when we try to use it In any other place in the codebase.

Will work on the 3 comments, thanks for the review 😄

File: src/uidl-definitions/types/index.ts (hunk 2 above)
Comment (JayaKrishnaNamburu) [reply src/uidl-definitions/types/index.ts]: It just holds the screenshots of the App but now i don't think it is not used much after [referring](https://www.w3.org/TR/appmanifest/#screenshots-member) to few examples, will remove that :D

File: src/uidl-definitions/types/index.ts (hunk 3 above)
Comment (JayaKrishnaNamburu) [reply src/uidl-definitions/types/index.ts]: Hey @alexnm i just pushed by making all the changes, please let me know if we go with this style of declaration in ts files so i will go ahead and update then and update all the files in codebase where we are using them.
~~~
export enum AssetType {
  "style",
}

Comment (alexnm) [reply src/uidl-definitions/types/index.ts]: @JayaKrishnaNamburu looks great, please go ahead and add the TS enums. you can also check in the codebase where the strings are used and replace them. curious if it has any impact in running the tests, but all the generators should behave the same after this change.

Comment (JayaKrishnaNamburu) [reply src/uidl-definitions/types/index.ts]: ok, will work changing them 👍

Comment (JayaKrishnaNamburu) [reply src/uidl-definitions/types/index.ts]: @alexnm i don't think we can go ahead with this approach, here is the challenge in changing it into enum in ts files.

All Enum's are converted into objects in return so we will be able to using as ``AssetType.style`` in the code base but which helps in eliminating strings, but when we can't using ``AssetType.style`` in our project UIDL schema right, so the generators are breaking 😅  I need your inputs 😅

Comment (JayaKrishnaNamburu) [reply src/uidl-definitions/types/index.ts]: declaring them in union style all the builds and projects are being generated without breaking

Comment (alexnm) [reply src/uidl-definitions/types/index.ts]: please go with the union types for now, we'll figure out later if it makes sense to use enums or not. the @babel/types library we are using for manipulating ASTs is also using a similar approach as far as I can see

**Comment by JayaKrishnaNamburu** [comment]
Hi @alexnm i just updated the PR, i just updated only the ``defaultValue`` for the types, because if i try to change the type here is the challenge i am facing currently.
In plugins like ``teleport-react-base-component` we have checks like this ``type: stateDefinitions[stateKey].type`` which is declared to check the type that is passed which is ``string`` so the ``generators`` and ``tests`` are breaking 😅  Please suggest your views on this.

**Comment by alexnm** [comment]
@JayaKrishnaNamburu let's leave it like that for now, I'll merge this , I can take a look at the rest of the types when I'm done with #83, which involves some changes into props and state anyway

**Comment by alexnm** [comment]
please take a look at #77 if you have time so we can also close that today

**Comment by JayaKrishnaNamburu** [comment]
> please take a look at #77 if you have time so we can also close that today

Hey, @alexnm you want me to try out Lerna or any other alternatives or just make a proper ``bump.json`` so we can merge into master?

**Comment by alexnm** [comment]
no, just check the comments left by @Utwo

**Pull request merged**